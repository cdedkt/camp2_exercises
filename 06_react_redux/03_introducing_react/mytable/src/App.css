.cell-fix {
  background-color: yellow;
}
