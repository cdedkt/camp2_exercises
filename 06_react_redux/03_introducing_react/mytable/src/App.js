import React, { Component } from 'react';
import './App.css';
import Cex from './Cex';

class CellCex extends Component {
  constructor(props) {
    super(props);
    //console.log("cell cex=", props.cex, ", indicator", props.indicator);
    this.state = {
      value: props.value,
    }
    this.handleChange = this.handleChange.bind(this);
    this.handleFocus = this.handleFocus.bind(this);
    this.handleBlur = this.handleBlur.bind(this);
  }

  componentWillReceiveProps(props) {
    this.setState({ value: props.value })
  }

  checkFormat(value) {
    //const re = /(\b[a-z](?!\s))/g;
    //return value.replace(re, function(x){return x.toUpperCase();});
    return value;
  }

  handleChange(event) {
    console.log("handleChange: event.target.id=",event.target.id);
    this.setState({value: this.checkFormat(event.target.value)});
  }

  handleFocus(event) {
    console.log("handleFocus: event.target.id=",event.target.id);
    this.setState({valueOnFocus: this.state.value});
  }

  handleBlur(event) {
    console.log("handleBlur: event.target.id=",event.target.id);
    if (this.state.valueOnFocus !== this.state.value) {
      console.log("handleBlur: value is modified");
      this.props.changeCexValue(event.target.id, this.state.value);
    } else {
      console.log("handleBlur: value is NOT modified");
    }
  }

  render() {
    return (
      this.props.indicator.modified
        ? <td className="cell-cex cell-cex-modified">
            <input
              id={Cex.generateKey(this.props.entity, this.props.period, this.props.indicator.name)}
              type="text"
              value={this.state.value}
              onChange={this.handleChange} onFocus={this.handleFocus} onBlur={this.handleBlur}/>
          </td>
        : <td className="cell-cex">
            <span id={Cex.generateKey(this.props.entity, this.props.period, this.props.indicator.name)}>
              {this.state.value}
            </span>
          </td>
    );
  }
}


class IndicatorRow extends Component {
  render() {
    return (
      <tr>
        <th scope="row">{this.props.indicator.name}</th>
        {this.props.cexList.map(currentCex =>
          <CellCex
            key={Cex.generateKey(currentCex.entity, currentCex.period, this.props.indicator.name)}
            entity={currentCex.entity}
            period={currentCex.period}
            value={currentCex[this.props.indicator.name]}
            indicator={this.props.indicator}
            changeCexValue={this.props.changeCexValue}/>
        )}
      </tr>
    );
  }
}


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      cexList: props.cexList,
    };
    this.changeCexValue = this.changeCexValue.bind(this);
  }

  changeCexValue(key, value) {
    const newCexList = Cex.modifyCexValue(this.state.cexList, key, value);

    this.setState({
      cexList: [...newCexList],
    });
  }

  render() {
    return (
      <div className="container-fluid">
        <table className="table">
          <thead>
            <tr>
              <th scope="col">#</th>
              {this.state.cexList.map(cex =>
                <th scope="col"><div>{cex.entity}</div><div>{cex.period}</div></th>)
              }
            </tr>
          </thead>
          <tbody>
            {this.props.indicatorList.map(indicator =>
              <IndicatorRow key={indicator.name} indicator={indicator} cexList={this.state.cexList} changeCexValue={this.changeCexValue}/>)
            }
          </tbody>
        </table>
      </div>
    );
  }
}

export default App;
