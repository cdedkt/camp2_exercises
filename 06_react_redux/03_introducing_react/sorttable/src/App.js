import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

import products from './dataProducts';
import TableProducts from './TableProducts';

class App extends Component {
  render() {
    return (
      <div>
        <TableProducts products={products}/>
      </div>
    );
  }
}

export default App;
