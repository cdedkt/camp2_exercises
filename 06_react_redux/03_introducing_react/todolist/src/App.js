import React, { Component } from 'react';
import './App.css';

import todos from './dataTodos';
import TableTodos from './TableTodos';


class App extends Component {


  render() {
    return (
      <div>
        <TableTodos todos={todos}/>
      </div>
    );
  }
}

export default App;
