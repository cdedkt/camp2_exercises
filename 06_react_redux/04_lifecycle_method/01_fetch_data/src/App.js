import React, { Component } from 'react';
import './App.css';

class DogApp extends Component {
  constructor(props) {
    super(props);
    this.state = {
      url: "",
    };
  }

  nextImage() {
    fetch(
      `https://random.dog/woof.json`,
      {method: "GET"}
    )
      .then((response) => response.json())
      .then((dog) => {
        console.log("dog.url=",dog.url);
        this.setState({
          url: dog.url,
        });
      })
      .catch((error) => {
        console.warn(error);
      });
  }

  render() {
    console.log("render");
    return (
      <div className="App">
        <br/>
        <img src={this.state.url}/>
        <br/>
        <br/>
        <button onClick={() => this.nextImage()}>next</button>
      </div>
    );
  }

  componentDidMount() {
    console.log("componentDidMount");
    this.nextImage();
  }
}

export default DogApp;
