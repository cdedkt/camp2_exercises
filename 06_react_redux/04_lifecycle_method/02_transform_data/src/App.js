import React, { Component } from 'react';
import './App.css';
import _ from "underscore"

function TableHeader() {
  return (
    <tr>
      <th scope="col">what has been paid</th>
      <th scope="col">by whom?</th>
    </tr>
  )
}

class TableRow extends Component {
  render() {
    return (
      <tr>
        <td scope="row">{this.props.what}</td>
        <td>{this.props.by}</td>
      </tr>
    )
  }
}

// App has `expenses` and `users` as props
class App extends Component {
  render() {

    return (
      <div className="App">
        <table className="table table-striped mt-4">
          <thead>
            <TableHeader/>
          </thead>
          <tbody>
          {this.props.expenses.map((expense, index) => {

            const userFound = _.findWhere(this.props.users, {id: expense.who});
            let username = "";
            console.log("userFound=", userFound);
            if (userFound !== undefined) {
              username = userFound.name;
            }

            return (<TableRow key={index}
              what={expense.what}
              by={username}
            />)
            }
          )}
          </tbody>
        </table>
      </div>
    );
  }
}

export default App;
