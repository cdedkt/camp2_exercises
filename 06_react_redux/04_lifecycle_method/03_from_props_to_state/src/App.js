import React, { Component } from 'react';
import './App.css';
import Book from "./Book"

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isbn: "978059651774",
    }
  }
  handleChange = (event) => {
    console.log("handleChange");
    this.setState({isbnTemp: event.target.value});
  }
  handleSearch = () => {
    console.log("handleSearch");
    this.setState({isbn: this.state.isbnTemp});
  }

  componentDidMount() {
    this.setState({isbnTemp: this.state.isbn});
  }

  render() {
    return (
      <div className="App">
        <input type="text" value={this.state.isbnTemp} onChange={this.handleChange} />
        <button onClick={this.handleSearch}>Search</button>
        <Book isbn={this.state.isbn} />
        <br/>
        <div>this.state.isbnTemp = {this.state.isbnTemp}</div>
        <div>this.state.isbn = {this.state.isbn}</div>
      </div>
    );
  }
}

export default App;
