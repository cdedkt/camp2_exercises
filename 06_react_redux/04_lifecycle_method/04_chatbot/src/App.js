import React, { Component } from 'react';
import './App.css';
import ChatBot from "./ChatBot"

class CustomTextInput extends React.Component {
  constructor(props) {
    super(props);
    // create a ref to store the textInput DOM element
    this.refDom = React.createRef();
    this.focusTextInput = this.focusTextInput.bind(this);
  }

  focusTextInput() {
    // Explicitly focus the text input using the raw DOM API
    // Note: we're accessing "current" to get the DOM node
    this.refDom.current.focus();
  }

  render() {
    // tell React that we want to associate the <input> ref
    // with the `textInput` that we created in the constructor
    return (
      <div>
        <input
          type="text"
          ref={this.refDom} />

        <input
          type="button"
          value="Focus the text input"
          onClick={this.focusTextInput}
        />
      </div>
    );
  }
}


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      displayChat: true
    }
  }
  render() {
    return (
      <div className="App">
      <CustomTextInput/>
        <button onClick={() =>
          this.setState({displayChat: !this.state.displayChat})
        }>Toggle Chat</button>
        {this.state.displayChat
          ? <ChatBot />
          : null
        }
      </div>
    );
  }
}

export default App;
