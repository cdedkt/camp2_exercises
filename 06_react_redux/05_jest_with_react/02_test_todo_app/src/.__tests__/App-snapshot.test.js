// App.test.js  avec Snapshot
import React from 'react';
import App from '../App.js';
import renderer from 'react-test-renderer';


test('App should contain a input', () => {
  const component = renderer.create(
    <App />
  );
  let tree = component.toJSON();
  //console.log("tree=", tree);
  expect(tree).toMatchSnapshot();
});

test('App should have a clikable form', () => {
  const component = renderer.create(
    <App />
  );
  let tree = component.toJSON();

  //console.log(">>>>>>", tree.children[0].children[0]);
  tree.children[0].children[0].props.onSubmit({preventDefault: ()=>{}});
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
