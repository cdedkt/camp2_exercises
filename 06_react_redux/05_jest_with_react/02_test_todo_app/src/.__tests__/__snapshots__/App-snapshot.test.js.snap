// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should contain a input 1`] = `
<div
  className="Todo"
>
  <div
    className="new-todo"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        id="inputNewTask"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
  <ul
    className="todos"
  />
</div>
`;

exports[`App should have a clikable form 1`] = `
<div
  className="Todo"
>
  <div
    className="new-todo"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        id="inputNewTask"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
  <ul
    className="todos"
  >
    <li
      className="task"
    >
      <span>
        
      </span>
      <input
        onChange={[Function]}
        type="checkbox"
        value={false}
      />
    </li>
  </ul>
</div>
`;
